curvefs/src/mds/fs_manager.cpp

```diff
@@ -808,15 +806,29 @@ int FsManager::IsExactlySameOrCreateUnComplete(const std::string& fsName,
         return google::protobuf::util::MessageDifferencer::Equals(lhs, rhs);
     };
 
-    auto checkFsInfo = [fsType, volumeInfoComparator](const FsDetail& lhs,
-                                                      const FsDetail& rhs) {
+    auto s3InfoComparator = [](common::S3Info lhs, common::S3Info rhs) {
+        // for compatible with old clients
```

Comment:
Please add more comments to explain scenarios that require compatibility, and renaming parameters because it is impossible to clearly distinguish which is the MDS side and which is the fs the current client is preparing to create.

Reply:
The parameters' meaning depends on the caller, If changing the parameters, the caller must pass the parameter in the correct order. Is there any need to make changes?

Reply:
> The parameters' meaning depends on the caller, If changing the parameters, the caller must pass the parameter in the correct order. Is there any need to make changes?

The parameters of a function are part of the calling convention and should not depend on the caller. The caller needs to pass in the parameters according to the parameters of the function as specified.

Reply:
done